### ISSUE TITLE
[Bug] Failing to install with pip because of recent change to setuptools

### ISSUE THREAD

**Issue by smiledelf** [dbt-labs/dbt-core#10495]
### Is this a new bug in dbt-core?

- [X] I believe this is a new bug in dbt-core
- [X] I have searched the existing issues, and I could not find an existing issue for this bug

### Current Behavior

pip install `dbt-core` (or `dbt-databricks`) will run into an import problem when it tries to install `logbook`.
`ModuleNotFoundError: No module named 'setuptools.command.test'`
`note: This error originates from a subprocess, and is likely not a problem with pip.`
`error: subprocess-exited-with-error`

After some investigation:

- dbt-core lists `logbook==1.5` as a requirement
- logbook 1.5 tries to import `setuptools.command.test` but fails as it doesn't exist
- setuptools recently removed test command: [commit](https://github.com/pypa/setuptools/commit/c405ac1bf29b945db9af7ba9b0dd77e4d871f72a), [pypa/setuptools/issues/4519](https://github.com/pypa/setuptools/issues/4519)

Note: the latest logbook version (1.7) installs successfully so it looks like it's been fixed on their end.

### Expected Behavior

`python3 -m pip install dbt-core` should install successfully.

### Steps To Reproduce

1. With python virtual env: `python3 -m venv venv`
2. Activate: `. venv/bin/activate`
3. `python3 -m pip install dbt-core`
4. Get the error.

### Relevant log output

```shell
`python3 -m pip install dbt-core`
Collecting dbt-core
  Downloading dbt_core-1.8.4-py3-none-any.whl.metadata (3.9 kB)
Collecting agate<1.10,>=1.7.0 (from dbt-core)
  Downloading agate-1.9.1-py2.py3-none-any.whl.metadata (3.2 kB)
Collecting Jinja2<4,>=3.1.3 (from dbt-core)
  Using cached jinja2-3.1.4-py3-none-any.whl.metadata (2.6 kB)
Collecting mashumaro<4.0,>=3.9 (from mashumaro[msgpack]<4.0,>=3.9->dbt-core)
  Using cached mashumaro-3.13.1-py3-none-any.whl.metadata (114 kB)
Collecting logbook<1.6,>=1.5 (from dbt-core)
  Using cached Logbook-1.5.3.tar.gz (85 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error

  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [20 lines of output]
      Traceback (most recent call last):

......

File "/tmp/pip-build-env-yxiain02/overlay/lib/python3.12/site-packages/setuptools/build_meta.py", line 327, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=[])
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-yxiain02/overlay/lib/python3.12/site-packages/setuptools/build_meta.py", line 297, in _get_build_requires
          self.run_setup()
        File "/tmp/pip-build-env-yxiain02/overlay/lib/python3.12/site-packages/setuptools/build_meta.py", line 497, in run_setup
          super().run_setup(setup_script=setup_script)
        File "/tmp/pip-build-env-yxiain02/overlay/lib/python3.12/site-packages/setuptools/build_meta.py", line 313, in run_setup
          exec(code, locals())
        File "<string>", line 64, in <module>
      ModuleNotFoundError: No module named 'setuptools.command.test'
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
```

### Environment

```markdown
- OS: Ubuntu 24.04 LTS
- Python:3.12.3
- dbt: 1.8
```

### Which database adapter are you using with dbt?

other (mention it in "Additional Context")

### Additional Context

databricks (trying to install `dbt-databricks`) but the same bug appears when installing `dbt-core`

**Comment by confidantduk** [comment]
workaround mentioned in [this thread](https://github.com/pypa/setuptools/issues/4519#issuecomment-2254983472)

**Comment by harukitake** [comment]
I think this problem will be solved if https://github.com/dbt-labs/dbt-core/pull/10119 is released

**Comment by harukitake** [comment]
I have the same issue.

environment
- Ubuntu 22.04
- python 3.12.0
- dbt-core 1.7.17

**Comment by juan-abia** [comment]
Same issue here:
- python 3.11.6
- dbt-core 1.8.4
- setuptools 71.0.3

**Comment by toadjaune** [comment]
> I think this problem will be solved if #10119 is released

I'm confused though, it has been merged since may

**Comment by stefanooldeman** [comment]
Perhaps something in poetry changed that invokes a different path during builds?

```
ChefBuildError
  Backend subprocess exited when trying to invoke get_requires_for_build_wheel
  
  Traceback (most recent call last):
    File "/opt/poetry/venv/lib/python3.11/site-packages/pyproject_hooks/_in_process/_in_process.py", line 373, in <module>
      main()
    File "/opt/poetry/venv/lib/python3.11/site-packages/pyproject_hooks/_in_process/_in_process.py", line 357, in main
      json_out["return_val"] = hook(**hook_input["kwargs"])
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "/opt/poetry/venv/lib/python3.11/site-packages/pyproject_hooks/_in_process/_in_process.py", line 134, in get_requires_for_build_wheel
      return hook(config_settings)
             ^^^^^^^^^^^^^^^^^^^^^
    File "/tmp/tmp4tuj0f54/.venv/lib/python3.11/site-packages/setuptools/build_meta.py", line 327, in get_requires_for_build_wheel
      return self._get_build_requires(config_settings, requirements=[])
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "/tmp/tmp4tuj0f54/.venv/lib/python3.11/site-packages/setuptools/build_meta.py", line 297, in _get_build_requires
      self.run_setup()
    File "/tmp/tmp4tuj0f54/.venv/lib/python3.11/site-packages/setuptools/build_meta.py", line 497, in run_setup
      super().run_setup(setup_script=setup_script)
    File "/tmp/tmp4tuj0f54/.venv/lib/python3.11/site-packages/setuptools/build_meta.py", line 313, in run_setup
      exec(code, locals())
    File "<string>", line 64, in <module>
  ModuleNotFoundError: No module named 'setuptools.command.test'
  
  at /opt/poetry/venv/lib/python3.11/site-packages/poetry/installation/chef.py:164 in _prepare
      [160](https://gitlab.bol.io/advertising/data-foundation/mad/madside/-/jobs/20133007#L160)│ 
      161│                 error = ChefBuildError("\n\n".join(message_parts))
      [162](https://gitlab.bol.io/advertising/data-foundation/mad/madside/-/jobs/20133007#L162)│ 
      163│             if error is not None:
    → [164](https://gitlab.bol.io/advertising/data-foundation/mad/madside/-/jobs/20133007#L164)│                 raise error from None
      [165](https://gitlab.bol.io/advertising/data-foundation/mad/madside/-/jobs/20133007#L165)│ 
      166│             return path
      [167](https://gitlab.bol.io/advertising/data-foundation/mad/madside/-/jobs/20133007#L167)│ 
      168│     def _prepare_sdist(self, archive: Path, destination: Path | None = None) -> Path:
```

> Note: This error originates from the build backend, and is likely not a problem with poetry but with logbook (1.5.3) not supporting PEP 517 builds. You can verify this by running 'pip wheel --no-cache-dir --use-pep517 "logbook (==1.5.3)"'.

Could we allow installation of logbook 1.6, it could have a fix already available:
https://github.com/getlogbook/logbook/blob/develop/CHANGES#L21

**Comment by junronglau** [comment]
Adding these before installation of dbt-core solved it for me

```
python -m pip install --upgrade pip-tools pip wheel
python -m pip install setuptools==71.1.0 poetry==1.8.3
```

**Comment by mkielar** [comment]
> > I think this problem will be solved if #10119 is released
> 
> I'm confused though, it has been merged since may, while the latest release is a week old

They haven't remove the dependency, though, @toadjaune. It's still there in 1.8.4.
https://github.com/dbt-labs/dbt-core/blob/v1.8.4/core/setup.py#L55-L58

**Comment by rbenomar** [comment]
> > > I think this problem will be solved if #10119 is released
> > 
> > 
> > I'm confused though, it has been merged since may, while the latest release is a week old
> 
> They haven't removed the dependency, though, @toadjaune. It's still there in 1.8.4. https://github.com/dbt-labs/dbt-core/blob/v1.8.4/core/setup.py#L55-L58

there's a dependabot pull request from a year ago:
https://github.com/dbt-labs/dbt-core/pull/8254

**Comment by D3nn3** [comment]
fyi: the "faulty" setuptools release was yanked and everything should work as expected again.

https://github.com/pypa/setuptools/issues/4519#issuecomment-2255940870
https://pypi.org/project/setuptools/#history

**Comment by CincyBC** [comment]
The `setuptools` package released 72.1.0 about an hour ago, which we were able to successfully use to install the Logbook dependency.
